src/gbm/gblinear.cc

```diff
@@ -73,9 +75,25 @@ class GBLinear : public GradientBooster {
   void Load(dmlc::Stream* fi) override {
     model_.Load(fi);
   }
+  void Load(Json const& in) override {
+    fromJson(in["gblinear_train_param"], &param_);
```

Comment:
Should we add an assert that in["name"] == "gblinear" ?

Reply:
@trams Since this is inside GBLinear, I don't think the object can be initialized in the first place if `name != gblinear`.